SFTP/SFTP.php

```diff
@@ -13,10 +13,16 @@ public function __construct()
     {
     }
 
-    public function connect($host, $username, $password)
+    public function connect($host, $username, $password=NULL)
     {
         $connection = ssh2_connect($host);
-        ssh2_auth_password($connection, $username, $password);
+
+        if ($password.is_null()) {
```

Comment:
This line will broke code because of following reasons:
1) PHP doesn't support dot notation for calls;
2) As $password is not object we can't call any function of it.

Please change this line to be:
 `if (is_null($password)) {`